crates/cairo-lint-core/src/lints/ifs/ifs_same_cond.rs

```diff
@@ -0,0 +1,54 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::{BlockOrIf, Condition, Expr, ExprIf, OptionElseClause};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::TypedSyntaxNode;
+
+pub const DUPLICATE_IF_CONDITION: &str = "Consecutive `if` with the same condition found.";
+
+fn is_function_call(expr: &Expr) -> bool {
+    matches!(expr, Expr::FunctionCall(_))
+}
```

Comment:
i don't think this is necessary you can directly use what's inside the function

Reply:
Do you mean using it like this 
```
      let is_if_function_call = match &if_condition {
                Condition::Expr(expr) =>  matches!(&expr.expr(db), Expr::FunctionCall(_)),
                Condition::Let(condition_let) => matches!(&condition_let.expr(db),Expr::FunctionCall(_)),
            };

            let is_else_if_function_call = match &else_if_condition {
                Condition::Expr(expr) => matches!(&expr.expr(db), Expr::FunctionCall(_)),
                Condition::Let(condition_let) => matches!(&condition_let.expr(db),Expr::FunctionCall(_)),
            };

            if is_if_function_call || is_else_if_function_call {
                return;
            }
```

Reply:
or do you like it better like this?
```
if matches!(&if_condition, Condition::Expr(expr) if matches!(expr.expr(db), Expr::FunctionCall(_)))
    || matches!(&if_condition, Condition::Let(condition_let) if matches!(condition_let.expr(db), Expr::FunctionCall(_)))
{
    return; // Salimos si la condición es una llamada a función
}
```

Reply:
Now that I think about it, I think we can only validate with the if if it is a function or not, right? because if the if is not a function but the else_if is, they would no longer match and it would not apply to the case